webapp/src/ts/services/target-aggregates.service.ts

```diff
@@ -295,11 +308,23 @@ export class TargetAggregatesService {
     return !facilityIds || facilityIds.length > 0;
   }
 
-  getAggregates(facilityId?) {
-    return this.ngZone.runOutsideAngular(() => this._getAggregates(facilityId));
+  getReportingMonth(reportingPeriod) {
+    return this.settingsService
+      .get()
+      .then(settings => {
+        const tag = reportingPeriod === ReportingPeriod.PREVIOUS
```

Comment:
This code is repeated, so better make a private `getTag` function.

Please add unit tests for the new functions